core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1230,6 +1260,18 @@ public Builder addSnapshot(Snapshot snapshot) {
       snapshotsById.put(snapshot.snapshotId(), snapshot);
       changes.add(new MetadataUpdate.AddSnapshot(snapshot));
 
+      if (rowLineage) {
+        ValidationException.check(
+            snapshot.firstRowId() >= lastRowId,
+            "Cannot add a snapshot whose first-row-id (%s) is less than the metadata `last-used-id` (%s) because this will end up generating duplicate row_ids.",
+            snapshot.firstRowId(),
+            lastRowId);
+        ValidationException.check(
+            snapshot.addedRows() != null,
+            "Cannot add a snapshot with a null `addedRows` field when row lineage is enabled");
+        this.incrementLastRowId(snapshot.addedRows());
```

Comment:
is this `this` intentional?

Reply:
i think we have inconsistent style around this. Sometimes in the file we use this.$accessor and other times just $accessor

Reply:
Tried to switch this out so all reads are thisless and all sets are thisfull